apis/v1alpha1/snapshot.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SnapshotSpec defines the desired state of Snapshot.
+//
+// Represents a copy of an entire cluster as of the time when the snapshot was
+// taken.
+type SnapshotSpec struct {
+	// The snapshot is created from this cluster.
+	ClusterName *string `json:"clusterName,omitempty"`
+	// The ID of the KMS key used to encrypt the snapshot.
+	KMSKeyID *string `json:"kmsKeyID,omitempty"`
+	// A name for the snapshot being created.
+	// +kubebuilder:validation:Required
+	SnapshotName *string `json:"snapshotName"`
```

Comment:
Can you please rename this just `Name`? Use the [S3 controller](https://github.com/aws-controllers-k8s/s3-controller/blob/main/generator.yaml#L103-L113) for reference